src/Routing/Middleware/AssetMiddleware.php

```diff
@@ -69,20 +48,17 @@ public function __construct(array $options = [])
         if (!empty($options['cacheTime'])) {
             $this->cacheTime = $options['cacheTime'];
         }
-        if (!empty($options['types'])) {
-            $this->typeMap = array_merge($this->typeMap, $options['types']);
-        }
     }
 
     /**
      * Serve assets if the path matches one.
      *
      * @param \Psr\Http\Message\ServerRequestInterface $request The request.
-     * @param \Psr\Http\Message\ResponseInterface $response The response.
+     * @param \Cake\Http\Response $response The response.
      * @param callable $next Callback to invoke the next middleware.
-     * @return \Psr\Http\Message\ResponseInterface A response
+     * @return \Cake\Http\Response A response
      */
-    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next): ResponseInterface
+    public function __invoke(ServerRequestInterface $request, Response $response, callable $next): Response
```

Comment:
Shouldnt that be our response interface then instead?

    ServerRequestInterface $request, ServerResponseInterface $response

would be a bit cleaner, wouldnt it?

Reply:
I personally think interfaces should be added only if there are going to be multiple implementations for it, not just to replace typehints.

Reply:
In light of psr comp. some might want to exchange with their or some other response class I would imagine.

Reply:
@ravage84 The getMimeType() method used on response instance in deliverAsset() method does not exist on the interface.